lib/driver.js

```diff
@@ -261,9 +261,15 @@ class XCUITestDriver extends BaseDriver {
       this.opts.bundleId = await this.extractBundleId(this.opts.app);
     }
 
-    if (!this.opts.realDevice && this.opts.scaleFactor !== undefined) {
-      log.info(`Settings non-default Simulator scale factor ${this.opts.scaleFactor}`);
-      device.setScaleFactor(this.opts.scaleFactor);
+    if (!this.opts.realDevice) {
+      if (typeof this.opts.scaleFactor !== 'undefined') {
+        log.info(`Setting non-default Simulator scale factor ${this.opts.scaleFactor}`);
+        device.setScaleFactor(this.opts.scaleFactor);
+      }
+      if (typeof this.opts.connectHardwareKeyboard !== 'undefined') {
+        log.info(`Setting connectHardwareKeyboard Simulator option to ${this.opts.connectHardwareKeyboard}`);
```

Comment:
Based on how the simulator is working, this is either on or off. This log line will be odd ("Setting connectHardwareKeyboard Simulator option to null", etc.). It would be better to be either "off" or "on" depending on the value.